### PR TITLE
Lt 4271 message logging

### PR DESCRIPTION
**Pull request by gponomarev-lykke**

### CODE REVIEW

File: src/MarginTrading.Backend.Services/Notifications/IRabbitMqProducerContainer.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        void RegisterProducer<TMessage>(RabbitMqQueueInfo queueInfo);
+
+        /// <summary>
+        /// Registers a producer with extended settings from RabbitMqSettings
+        /// Human-readable message logging is enabled based on the shouldLogProducedEvents flag
+        /// </summary>
+        /// <param name="settings"></param>
+        /// <param name="shouldLogProducedEvents"></param>
+        /// <typeparam name="TMessage"></typeparam>
+        void RegisterProducer<TMessage>(RabbitMqSettings settings, bool shouldLogProducedEvents);
```
Comment (tarurar) [inline src/MarginTrading.Backend.Services/Notifications/IRabbitMqProducerContainer.cs:34]: I am confused, what is the difference with `void RegisterProducer<TMessage>(RabbitMqQueueInfoWithLogging queueInfo);` method?

File: src/MarginTrading.Backend.Services/Notifications/RabbitMqNotifyService.cs [hunk 2]
```diff
@@ -55,106 +54,88 @@ public Task OrderHistory(Order order, OrderUpdateType orderUpdateType, string ac
...
             try
             {
-                messageStr = message.ToJson();
-                await _publishers[exchangeName].ProduceAsync(messageStr);
+                var (queueInfo, producer) = _producerContainer.GetProducer<T>();
+                exchangeName = queueInfo.ExchangeName;
+                await producer.ProduceAsync(message);
 
-                if (logEvent)
+                if (queueInfo.LogEventPublishing)
```
Comment (tarurar) [inline src/MarginTrading.Backend.Services/Notifications/RabbitMqNotifyService.cs:115]: Why not use a decorator for container or producer for this logging concerns?

File: src/MarginTrading.Backend.Services/Notifications/IRabbitMqProducerContainer.cs (hunk 1 above)
Comment (gponomarev-lykke) [reply src/MarginTrading.Backend.Services/Notifications/IRabbitMqProducerContainer.cs]: fixed

File: src/MarginTrading.Backend.Services/Notifications/RabbitMqNotifyService.cs (hunk 2 above)
Comment (gponomarev-lykke) [reply src/MarginTrading.Backend.Services/Notifications/RabbitMqNotifyService.cs]: This is an older version of logging, specific only to MT Core (as opposed to a generic version that was introduced in this task). I decided not to refactor to much to save time and avoid bugs.

File: src/MarginTrading.Backend.Services/Notifications/RabbitMqProducerContainer.cs
Comment (tarurar) [inline src/MarginTrading.Backend.Services/Notifications/RabbitMqProducerContainer.cs:1]: Why not use DI container for this purpose?

Comment (gponomarev-lykke) [reply src/MarginTrading.Backend.Services/Notifications/RabbitMqProducerContainer.cs:1]: I don't remember the exact reasoning, but perhaps the DI configuration would be to convoluted in such case. Usually the complexity of configuration for a collection of generic services rises up significantly. I think the idea was to implement it in way that's simple to reason about without reading lots of docs on DI configuration

**Review by tarurar — APPROVED** [review]

**Pull request merged**